cse/indexing/InvertedIndexReader.py

```diff
@@ -0,0 +1,34 @@
+import os
+
+from cse.indexing.Dictionary import Dictionary
+from cse.indexing.MainPostingListIndex import MainPostingListIndex
+
+
+class InvertedIndexReader(object):
+
+
+    __dictionary = None
+    __dIndex = None
+    __mIndex = None
+
+
+    def __init__(self, filepath):
+        self.__dictionary = Dictionary(os.path.join(filepath, "dictionary.index"))
+        self.__mIndex = MainPostingListIndex(os.path.join(filepath, "postingLists.index"))
+
+
+    def close(self):
+        self.__dictionary.close()
+        self.__mIndex.close()
+
+
+    def retrieve(self, term):
+        if term in self.__dictionary:
+            pointer = self.__dictionary[term]
+            return self.__mIndex[pointer]
```

Comment:
Out of curiosity: How is it implemented that you can call `[]` on the `MainPostingListIndex`-Class?

Reply:
Python has method contracts for implementing container types, see https://docs.python.org/3/reference/datamodel.html?emulating-container-types#emulating-container-types
eg. I used: `__getitem__` or `__setitem__`